packages/entity-browser/src/util/reduxForms.js

```diff
@@ -1,17 +1,30 @@
 const wholeEntityField = '___entity'
 import {fetchRequest} from './rest'
 import {SubmissionError} from 'redux-form'
+import _reduce from 'lodash/reduce'
+import _isEqual from 'lodash/isEqual'
 
-export const formValuesToEntity = values => {
+export const formValuesToEntity = (values, dirtyFields) => {
   const entity = values[wholeEntityField]
+  const {model, version, key} = entity
+
+  const result = {model, version, key, paths: {}}
+
+  const ignoreField = fieldName => (fieldName === wholeEntityField || (dirtyFields && !dirtyFields.includes(fieldName)))
 
   Object.keys(values).forEach(key => {
-    if (key !== wholeEntityField) {
-      entity.paths[key].value.value = values[key]
+    if (!ignoreField(key)) {
+      const type = entity.paths[key].type
+      if (type === 'field') {
+        result.paths[key] = values[key]
+      } else if (type === 'entity') {
+        result.paths[key] = {key: values[key]}
+      } else if (type === 'entity-list') {
```

Comment:
this will probably change. but we can have a look at it again as soon as the backend is ready.